### PR TITLE
Other Gases on hover

### PR DESCRIPTION
**Pull request by PrateekGupta1509**
- Added emission of CH4, N2O, GHG
- Couldn't find data for electric fuel. So added error handling for that.
- Error handling for tree string.

### CODE REVIEW

File: Source/core/resources/fuels.json [hunk 1]
```diff
@@ -3,67 +3,100 @@
     "langKey":"fuelDiesel",
     "measuredBy":"v",
     "CO2Emission":"2.614",
+    "CH4Emission":"0.00011095",
```
Comment (Ceilican) [inline Source/core/resources/fuels.json]: Could you add a comment pointing to the source of these values?

**Comment by Ceilican** [comment]
Thanks!

File: Source/core/resources/fuels.json [hunk 2]
```diff
@@ -3,67 +3,100 @@
...
-     "_comments":"kg/L;(0.8 * 2.614) + (0.20 * 2.486) = 2.5884"
+    "CH4Emission":"0.00009615",
+    "N2OEmission":"0.000017432",
+    "GHGEmission":"2.6795",
+     "_comments":"kg/L;(0.8 * diesel) + (0.20 * Bio-Diesel)"
   },
   "LPG":{
     "langKey":"fuelLPG",
-    "measuredBy":"m",
+    "measuredBy":"v",
```
Comment (heychirag) [inline Source/core/resources/fuels.json]: Isn't LPG a compressed gas?

**Comment by PrateekGupta1509** [comment]
LPG is liquefied petroleum gas. So its measured by volume. Also in the PDF which I used as source measured emission in per gallon [link here].(https://www.epa.gov/sites/production/files/2015-11/documents/emission-factors_nov_2015.pdf)

**Comment by heychirag** [comment]
@PrateekGupta1509, agreed. But then why is CNG sold by mass? It's also gas composed primarily of methane. Can you check this?

**Comment by PrateekGupta1509** [comment]
@heychirag 

>  The density of LNG is roughly 0.41 kg/liter to 0.5 kg/liter, depending on temperature, pressure, and composition,[3] compared to water at 1.0 kg/liter. Using the median value of 0.45 kg/liter
Source: https://en.wikipedia.org/wiki/Liquefied_natural_gas

It seems an average is taken for it to disregard the variations. Also from what I know, for CNG since the pressure of the compressor varies at the CNG station, a constant mass is better to use which remains same regardless of volume and pressure of container.

**Comment by heychirag** [comment]
The article is interesting. Thanks for elucidating @PrateekGupta1509. The only thing that worries me is I think you have mistaken LNG for LPG. These two are entirely different mixture of gases. Refer [this article](http://www.afsglobal.com/faq/gas-comparisons.html).

[This article](http://www.elgas.com.au/blog/389-lpg-conversions-kg-litres-mj-kwh-and-m3) says it's 0.51 kg/litre for LPG. So there is bit confusion regarding the exact conversion rate.

**Pull request merged**